### PR TITLE
GH-12808: Update migration doc

### LINKED ISSUE #12808: Improve checkout lib migration doc

### PR DESCRIPTION
**Pull request by plabadie**

### CODE REVIEW

File: docs/migration/4_0.md [hunk 1]
```diff
@@ -253,14 +253,40 @@ Lib: @spartacus/core
...
 Change: Two new required constructor parameters `userAddressConnector: UserAddressConnector` and `command: CommandService`
 
+
+## New Checkout Library
+
+Spartacus 4.0 introduces the checkout library.  The checkout related code is moved out of `@spartacus/core` and `@spartacus/storefrontlib` into one of the checkout lib's entry points.  The checkout library is split into these entry points:
+
+```
+@spartacus/checkout/assets 
+( checout related i18n keys are moved here )
```
Comment (WeizhengSap) [inline docs/migration/4_0.md]: typo `checout`

File: docs/migration/4_0.md [hunk 2]
```diff
@@ -253,14 +253,40 @@ Lib: @spartacus/core
...
+
+@spartacus/checkout/components
+( checout related UI codee is moved here. This includes components, guards anD ui services )
```
Comment (WeizhengSap) [inline docs/migration/4_0.md]: typo `checout` and `codee`

File: docs/migration/4_0.md [hunk 3]
```diff
@@ -253,14 +253,40 @@ Lib: @spartacus/core
...
+
+@spartacus/checkout/core
+The checkout facade API implementation are moved here, as well as connectors, event builder, event listener, models, other services, and state management.
+
+@spartacus/checkout/occ
+The checkout related OCC code is moved here. This includes the checkout related adapters and converters.
+
+@spartacus/checkout/root
+The root entry point is, by convention, meant to always be eager loaded.  It contains the config, events, facades, http interceptors and models.
```
Comment (WeizhengSap) [inline docs/migration/4_0.md:311]: facade interfaces, right? the implementation still in `checkout core`

File: docs/migration/4_0.md (hunk 1 above)
Comment (plabadie) [reply docs/migration/4_0.md]: Thanks, fixed

File: docs/migration/4_0.md (hunk 2 above)
Comment (plabadie) [reply docs/migration/4_0.md]: Fixed

File: docs/migration/4_0.md (hunk 3 above)
Comment (plabadie) [reply docs/migration/4_0.md:311]: This is somewhat hidden for those who use the libs.  Their dependency for facades should be diirectly on the facades from the root entry point and they behave as if they were full iimplementations even though they are abrtract classe proxies.

**Review by WeizhengSap — APPROVED** [review]

**Pull request merged**